lib/manageiq/appliance_console/cli.rb

```diff
@@ -344,6 +368,33 @@ def set_replication
       db_replication.activate
     end
 
+    def db_dump
+      PostgresAdmin.backup_pg_dump(extract_db_opts(options))
+    end
+
+    def db_backup
+      PostgresAdmin.backup(extract_db_opts(options))
+    end
+
+    def db_restore
+      PostgresAdmin.restore(extract_db_opts(options))
+    end
+
+    DB_OPT_KEYS = %i[dbname username password hostname port local_file].freeze
+    def extract_db_opts(options)
+      require 'manageiq/appliance_console/postgres_admin'
+
+      db_opts = {}
+
+      DB_OPT_KEYS.each { |k| db_opts[k] = options[k] if options[k] }
```

Comment:
Not sure if slice is available, but is this simpler?  If it's really important that keys without values are removed you could also tack on a `.compact` or `.delete_blanks` or something like that.

```suggestion
      db_opts = options.slice(*DB_OPT_KEYS)
```